src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -0,0 +1,151 @@
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            // warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+            // warning CS8600: Converting null literal or possible null value to non-nullable type.
+            get { return ImmutableArray.Create("CS8625", "CS8600"); }
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = editor.OriginalRoot;
+
+            // a method can have multiple `return null;` statements, but we should only fix its return type once
+            var alreadyHandled = new HashSet<TypeSyntax>();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                MakeDeclarationNullable(document, editor, node, alreadyHandled);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private void MakeDeclarationNullable(Document document, SyntaxEditor editor, SyntaxNode node, HashSet<TypeSyntax> alreadyHandled)
+        {
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix != null && !alreadyHandled.Contains(declarationTypeToFix))
+            {
+                var fixedDeclaration = SyntaxFactory.NullableType(declarationTypeToFix);
+                editor.ReplaceNode(declarationTypeToFix, fixedDeclaration);
+                alreadyHandled.Add(declarationTypeToFix);
+            }
+        }
+
+        private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
+        {
+            if (!node.IsKind(SyntaxKind.NullLiteralExpression))
+            {
+                return null;
+            }
+
+            if (node.IsParentKind(SyntaxKind.ReturnStatement))
+            {
+                var containingMember = node.GetAncestors().FirstOrDefault(a => a.IsKind(SyntaxKind.MethodDeclaration, SyntaxKind.PropertyDeclaration));
```

Comment:
what if the containing member is a local function?

Reply:
It'll be ignored for now. There is a tracking bug.

---
In reply to: [186234771](https://github.com/dotnet/roslyn/pull/26630#discussion_r186234771) [](ancestors = 186234771)

Reply:
Note: i would not add skipped tests for this.  Just add the tests with the current bahvior and the note that this will change when the corresponding bug is fixed.  That way they do break and are appropriately updated when the change is made .  Otherwise, they can just sit there skipped with people not noticing for years.

Reply:
@CyrusNajmabadi I'm sorry, that is kind of my fault if you look at the feedback above :smile:

Reply:
@jcouv I'm sorry to bother you about that :-/

Reply:
My reasoning was that regular tests with WorkAttribute would go unnoticed without raising any eyebrows as opposed to skipped tests. Sorry again you had to change this twice.

Reply:
No worries. Both rationales offer some benefits. Cyrus's argument (easier to forget skipped tests) carried the day ;-)